Cargo.toml

```
@@ -16,15 +16,15 @@ readme = "README.md"
 bincode = "1.3"
 either = "1.8"
 fnv = "1.0"
-itertools = "0.10"
+itertools = "0.11"
```

[MSRV seems to be 1.62](https://github.com/rust-itertools/itertools/blob/master/.github/workflows/ci.yml), I would love if we could keep 1.52 here for a while longer.

```suggestion
itertools = ">=0.10, <=0.11"
```

That should actually work for both cases, I think.